src/box/sql.c

```diff
@@ -1476,3 +1477,94 @@ index_field_tuple_est(const struct index_def *idx_def, uint32_t field)
 	}
 	return tnt_idx->def->opts.stat->tuple_log_est[field];
 }
+
+/**
+ * Find the constraint in space and return its JSON path in buffer allocated on
+ * region.
+ */
+static const char *
+sql_constraint_path(struct region *region, uint32_t space_id, const char *name)
+{
+	struct space *space = space_by_id(space_id);
+	if (space == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_SPACE, space_name);
+		return NULL;
+	}
+	uint32_t size = strlen(name) + 32;
+	char *path = region_alloc(region, size);
+	if (path == NULL) {
+		diag_set(OutOfMemory, size, "region_alloc", "path");
+		return NULL;
+	}
+	struct tuple_constraint_def *cdefs = space->def->opts.constraint_def;
+	uint32_t count = space->def->opts.constraint_count;
+	for (uint32_t i = 0; i < count; ++i) {
+		if (strcmp(cdefs[i].name, name) != 0)
+			continue;
+		enum tuple_constraint_type type = cdefs[i].type;
+		uint32_t type_count = 0;
+		for (uint32_t j = 0; j < count; ++j) {
+			if (cdefs[j].type == type)
+				++type_count;
+		}
+		assert(type_count > 0);
+		const char *type_str = tuple_constraint_type_strs[type];
+		if (type_count == 1)
+			snprintf(path, size, "flags.%s", type_str);
+		else
+			snprintf(path, size, "flags.%s.%s", type_str, name);
+		assert(strlen(path) < size);
```

Comment:
`strlen(path)` will always be less than `size` no matter what the size is - this is how `snprintf` works. You need to check the `snprintf` return value.

Reply:
Fixed.